### PR TITLE
Turn on id as any

### PR DESCRIPTION
**Pull request by jckarter**
Make -enable-id-as-any the default, and update standard library and overlays to match.

### CODE REVIEW

**Comment by jckarter** [comment]
@parkera This isn't done yet, but you might want to start looking over my overlay changes to verify that the places I expanded AnyObject to Any make sense.

**Comment by jckarter** [comment]
@swift-ci Please test

**Comment by jckarter** [comment]
@swift-ci Please test OS X

**Comment by jckarter** [comment]
@swift-ci Please test

**Comment by jckarter** [comment]
@swift-ci Please test Linux

File: stdlib/public/SDK/Foundation/Notification.swift [hunk 1]
```diff
@@ -31,15 +31,15 @@ public struct Notification : ReferenceConvertible, Equatable, Hashable {
...
+    public var object: Any?
     
     /// Storage for values or objects related to this notification.
     public var userInfo: [String : Any]?
     
     /// Initialize a new `Notification`.
     ///
     /// The default value for `userInfo` is nil.
-    public init(name: Name, object: AnyObject? = nil, userInfo: [String : Any]? = nil) {
+    public init(name: Name, object: Any? = nil, userInfo: [String : Any]? = nil) {
```
Comment (parkera) [inline stdlib/public/SDK/Foundation/Notification.swift]: We did some work here to create a struct Notification that could store any and be properly bridged to NSNotification. I wonder if it is no longer required, or we can leverage the work you've done.

Also, see this radar: 27426757

**Comment by parkera** [comment]
overlay diffs look ok to me so far, with the one note on struct Notification.

**Comment by jckarter** [comment]
@swift-ci Please test OS X

**Comment by jckarter** [comment]
@swift-ci Please test OS X

**Comment by jckarter** [comment]
@swift-ci Please test

**Comment by jckarter** [comment]
@swift-ci Please test

**Comment by jckarter** [comment]
@swift-ci please test linux

**Comment by jckarter** [comment]
@swift-ci Please test

**Comment by jckarter** [comment]
@swift-ci Please test and merge

**Comment by jckarter** [comment]
@parkera corelibs Foundation seems to have some unrelated migration issues:

```
TestFoundation/TestNSOrderedSet.swift:347:44: warning: expected 'let' in conditional
            if let lhs = lhs as? NSString, rhs = rhs as? NSString {
                                           ^
                                           let 
TestFoundation/TestNSOrderedSet.swift:358:44: warning: expected 'let' in conditional
            if let lhs = lhs as? NSString, rhs = rhs as? NSString {
                                           ^
                                           let 
TestFoundation/TestNSRegularExpression.swift:221:40: warning: expected ',' joining parts of a multi-clause condition
            if let first = firstResult where matches.count > 0 {
                                       ^~~~~
                                       ,
```

**Comment by jckarter** [comment]
@swift-ci Please test

**Comment by jckarter** [comment]
@parkera This shouldn't otherwise have much direct impact on corelibs, though of course any downstream projects attempting to be portable across Darwin and corelibs will hurt until corelibs' interface is updated to match. You think it's OK to land this for Darwin in that state?

**Comment by jckarter** [comment]
@gribozavr Thanks for confirming. Perhaps `Any` bridging interferes with the return-autoreleased optimization. I'll look into that.

File: stdlib/public/SDK/Foundation/Notification.swift (hunk 1 above)
Comment (jckarter) [inline stdlib/public/SDK/Foundation/Notification.swift]: @parkera Yeah, I noticed that as I was migrating. We could probably do away with that special handling now that the runtime can bridge anything for you.

Comment (jckarter) [inline stdlib/public/SDK/Foundation/Notification.swift]: 27426757 should also be addressed now that @rjmccall and @dabrahams have generalized collection casting.

**Comment by jckarter** [comment]
@parkera I updated the test in swift-corelibs-foundation. Let me try the tests again.

**Comment by jckarter** [comment]
@swift-ci Please test Linux

**Comment by parkera** [comment]
Thanks.

**Comment by jckarter** [comment]
@swift-ci Please test Linux

**Comment by atrick** [comment]
[18/18] SwiftExecutable: ../buildbot_linux/foundation-linux-x86_64/TestFoundation/TestFoundation
.../TestFoundation
.swift/utils/build-script-impl: line 235: 25910 Illegal instruction     "$@"

**Comment by jckarter** [comment]
@atrick Looking into it, thanks. When working through the stdlib this was often due to an unsafeBitCast where the type sizes no longer matched up. I wouldn't expect that to come up on Linux, though.

**Comment by jckarter** [comment]
The crash in Foundation seems to be due to a wanton array conversion, from array-of-unlabeled-tuple to array-of-tuple. I spoke with @rjmccall and we'll try tightening the conversions a bit.

**Comment by jckarter** [comment]
@swift-ci Please test

**Pull request merged**